subsys/settings/src/settings_fcb.c

```diff
@@ -79,52 +75,102 @@ int settings_fcb_dst(struct settings_fcb *cf)
 	return 0;
 }
 
-static int settings_fcb_load_cb(struct fcb_entry_ctx *entry_ctx, void *arg)
+/**
+ * @brief Check if there is any duplicate of the current setting
+ *
+ * This function checks if there is any duplicated data further in the buffer.
+ *
+ * @param cf        FCB handler
+ * @param entry_ctx Current entry context
+ * @param name      The name of the current entry
+ *
+ * @retval <0 Negative error code
+ * @retval 0  No duplicates found
+ * @retval >0 Duplicate found
+ */
+static int settings_fcb_check_duplicate(struct settings_fcb *cf,
+					struct fcb_entry_ctx *entry_ctx,
+					const char * const name)
 {
-	struct settings_fcb_load_cb_arg *argp;
-	char buf[SETTINGS_MAX_NAME_LEN + SETTINGS_EXTRA_LEN + 1];
+	struct fcb_entry_ctx entry2_ctx = *entry_ctx;
 	int rc;
 
-	argp = (struct settings_fcb_load_cb_arg *)arg;
-
-	size_t len_read;
+	while ((rc = fcb_getnext(&cf->cf_fcb, &entry2_ctx.loc)) == 0) {
+		char name2[SETTINGS_MAX_NAME_LEN + SETTINGS_EXTRA_LEN + 1];
+		size_t name2_len;
 
-	rc = settings_line_name_read(buf, sizeof(buf), &len_read,
-				     (void *)&entry_ctx->loc);
-	if (rc) {
-		return 0;
+		rc = settings_line_name_read(name2, sizeof(name2), &name2_len,
+						     &entry2_ctx);
+		name2[name2_len] = '\0';
+		if (rc) {
+			return rc;
+		}
+		if (!strcmp(name, name2))
```

Comment:
strcmp is not sufficient, suppose name ="bt/device" and name2="bt/device1"

Reply:
strcmp("bt/device", "bt/device1") != 0 - conditional part not executed. What's the problem?